metric_providers/psu/energy/ac/mcp/machine/source.c

```diff
@@ -241,11 +244,16 @@ int main(int argc, char **argv) {
     setvbuf(stdout, NULL, _IONBF, 0);
 
     fd = f511_init("/dev/ttyACM0");
+
     if(fd < 0) {
         fprintf(stderr, "Error. Connection could not be opened\n");
         return -1;
     }
 
+    if(check_system_flag && fd >= 0) {
```

Comment:
This check for fd >= 0 is redundant and I feel rather confusing. Make it rather an if / elseif clause with the one above.